src/options/output.rs

```diff
@@ -111,37 +89,54 @@ impl RoverPrinter for RoverOutput {
 }
 
 impl RoverPrinter for RoverError {
-    fn write_or_print(self, format_type: FormatType, _: FinalOutputType) -> RoverResult<()> {
-        match format_type {
-            FormatType::Plain => self.print(),
-            FormatType::Json => JsonOutput::from(self).print(),
+    fn write_or_print(self, output_opts: &OutputOpts) -> RoverResult<()> {
+        let (format_kind, output_destination) = output_opts.get_format_and_strategy();
+        match format_kind {
+            RoverOutputFormatKind::Plain => self.print(),
+            RoverOutputFormatKind::Json => {
+                let json = JsonOutput::from(self);
+                match output_destination {
+                    RoverOutputDestination::File(file) => {
+                        let success_heading = Style::Heading.paint(format!(
+                            "{}Error JSON was printed to",
+                            Emoji::Memo,
+                        ));
+                        Fs::write_file(&file, json.to_string())?;
+                        stderrln!("{} {}", success_heading, file)?;
+                    },
+                    RoverOutputDestination::Stdout => json.print()?
```

Comment:
In our pairing session we said we would always just print errors to `stderr`, but i realized that in the case where a file is passed AND json is passed that we should be writing that json out to a file instead of printing it to the terminal

Reply:
ooOOooooohhh icicic